posts/backstage-with-pechersky.md

```diff
@@ -0,0 +1,576 @@
+in 4 slightly incompatible ways. So mathlib also lets us explore different 
+usage and design strategies.
+
+The permutations work came out of trying to formalize Bernard Gross's elementary course
+on group theory that is available online. But that was hard because I was trying
+to make the LaTeX of the notes at the same time and show how to make such statements
+in mathlib. So that kind of petered out because we already have all of this in
+mathlib. Kevin Buzzard's "formalizing mathematics" course online is a much better
+version of what I was trying to do.
+
+But I do have an interest in p-adic numbers. Rob Lewis has a great paper out
+on his formalization of the p-adic's based on Gouvea's text -- Rob recommended
+this text to me during Lean for the Curious Mathematician in 2020. So while Rob
+has put high powered results in mathlib, I am just trying to start from the beginning
+and go through line by line, making sure that every statement has a clean
+way of being expressed and proven in mathlib. Especially around the "motivating"
+material, like, what is the expansion of `X / X + 1`. 
+
+I have this as a personal mathlib-dependent project: `pechersky/gouvea`. You
+can kind of see why my PRs are the way they are, you can just follow the book
+and see my trajectory of PRs -- proving Taylor expansions of a polynomial
+are what you expect they are, connecting fraction fiels and polynomials.
+This is all based on your (Johan Commelin) work defining Taylor
+via the cool Hasse derivative formalization, and with Damiano Testa's
+powerful polynomial lemmas that hide a lot of the complexity. And Anne Baanen's
+work on defining `ratfunc`. So I'm really standing on the shoulders of many other
+people.
+
+JMC: So you are going sentence-by-sentence through this book,
+and making sure that everything can be expressed in Lean.
+
+YP: Right, because thinking about pedagogy,
+we'd love to have these documents that someone can just click and explore.
+But for me, it's also been great to learn about ring theory and field theory and localizations.
+It's been great to talk with others in the community to learn more about this.
+
+And then it can lead into rabbit holes like non-commutative ring theory.
+Or another example:
+the proof that the ring of power series is Noetherian is slightly different from the proof
+that polynomial rings are Noetherian.
+And we don't have it yet; so maybe that's something I will do.
+I've just sort of been exploring these unfinished corners.
+
+With rational functions, I find it particularly interesting, because we have a diamond.
+From polynomials, you can go to rational functions, and then to Laurent series.
+Or you can go from polynomials to power series, and then to Laurent series.
+
+Anyway, long story short:
+I want to take rational functions, and be able to talk about roots, and poles, etc.
+And also connect this corner of algebra to analysis,
+to the work that people liky Yury Kudryashov and Patrick Massor have been doing.
```

Comment:
```suggestion
to the work that people like Yury Kudryashov and Patrick Massot have been doing.
```

Reply:
Be careful that Johan fixed only on of those two typos.